src/stash/mod.rs

```diff
@@ -268,7 +279,7 @@ impl<V> Stash<V> {
     ///
     /// *Panics* if the size of the `Stash<V>` would overflow `usize::MAX`.
     #[inline]
-    pub fn put(&mut self, value: V) -> usize {
+    pub fn put(&mut self, value: V) -> Ix {
         let loc = self.next_free;
```

Comment:
I'd prefer to create the index here (to panic early if necessary): `let idx = Ix::from_usize(self.next_free)`